client/state/stats/lists/utils.js

```diff
@@ -121,6 +121,53 @@ export const normalizers = {
 	},
 
 	/**
+	 * Returns a normalized payload from `/sites/{ site }/stats/top-posts`
+	 *
+	 * @param  {Object} data    Stats data
+	 * @param  {Object} query   Stats query
+	 * @return {Object?}        Normalized stats data
+	 */
+	statsTopPosts: ( data, query ) => {
+		if ( ! data || ! query.period || ! query.date ) {
+			return [];
+		}
+
+		const { startOf, endOf } = rangeOfPeriod( query.period, query.date );
+		const dataPath = query.summarize ? [ 'summary', 'postviews' ] : [ 'days', startOf, 'postviews' ];
+		const viewData = get( data, dataPath, [] );
+
+		return map( viewData, ( item ) => {
+			const detailPage = `/stats/post/${ item.id }/${ query.domain }`;
+			let inPeriod = false;
+
+			// Archive and home pages do not have dates
+			if ( item.date ) {
+				const postDate = moment( item.date );
+				// TODO: might be nice to update moment and use isSameOrAfter and isSameOrBefore
+				if (
+					( postDate.isAfter( startOf, 'day' ) || postDate.isSame( startOf, 'day' ) ) &&
+					( postDate.isBefore( endOf, 'day' ) || postDate.isSame( endOf, 'day' ) )
+				) {
+					inPeriod = true;
+				}
+			}
+
+			return {
+				label: item.title,
+				value: item.views,
+				page: detailPage,
+				actions: [ {
+					type: 'link',
+					data: item.href
+				} ],
+				labelIcon: null,
+				children: null,
+				className: inPeriod ? 'published' : null
```

Comment:
I'm wondering if the items not in the period are used later, or they're just hidden?